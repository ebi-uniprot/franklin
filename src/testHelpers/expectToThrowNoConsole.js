/* eslint-disable no-console */
const expectToThrowNoConsole = (func) => {
  // The error still gets printed to the console
  // so mock it before throwing the error
  jest.spyOn(console, 'error');
  console.error.mockImplementation(() => {});

  expect(func).toThrow();

  console.error.mockRestore();
};

export default expectToThrowNoConsole;
