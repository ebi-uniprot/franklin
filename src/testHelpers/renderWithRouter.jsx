import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

export default (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
