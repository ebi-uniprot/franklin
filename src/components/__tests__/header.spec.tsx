import Header from '../header';
import renderWithRouter from '../../testHelpers/renderWithRouter';
import expectToThrowNoConsole from '../../testHelpers/expectToThrowNoConsole';

describe('Header component', () => {
  test('should render', () => {
    const { asFragment } = renderWithRouter(
      <Header
        items={[
          { path: '/here', label: <strong>here</strong> },
          { path: '/there', label: 'there' },
          { href: 'www.ebi.ac.uk', label: 'EBI' },
          { onClick: () => {}, label: 'action' },
          {
            items: [
              { path: '/sub-link-a', label: 'sub link A' },
              { path: '/sub-link-b', label: <span>sub link b</span> },
            ],
            label: 'sub links',
          },
        ]}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
