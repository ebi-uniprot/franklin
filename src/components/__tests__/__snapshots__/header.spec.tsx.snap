// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header__logo"
    >
      <a
        href="/"
      />
    </div>
    <ul
      class="header__navigation"
    >
      <li>
        <a
          class="button tertiary"
          href="/here"
        >
          <strong>
            here
          </strong>
        </a>
      </li>
      <li>
        <a
          class="button tertiary"
          href="/there"
        >
          there
        </a>
      </li>
      <li>
        <a
          class="button tertiary"
          href="www.ebi.ac.uk"
          rel="noopener noreferrer"
          target="_blank"
        >
          EBI
        </a>
      </li>
      <li>
        <button
          class="button tertiary"
          type="button"
        >
          action
        </button>
      </li>
      <li>
        <div
          class="dropdown-container"
        >
          <button
            class="button primary dropdown"
            type="button"
          >
            sub links
          </button>
          <div
            class="dropdown-menu"
          />
        </div>
      </li>
    </ul>
    <div
      class="header__search"
    >
      <form
        class="main-search"
        data-testid="main-search-form"
        style="--main-button-color: #00639a;"
      >
        <div
          class="main-search__input"
        >
          <input
            aria-label="Text query"
            class="main-search__input"
            data-testid="main-search-input"
            type="text"
            value=""
          />
        </div>
        <button
          class="button primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="header__secondary"
    >
      <ul
        class="header__navigation"
      >
        <li>
          <a
            class="button tertiary"
            href="/"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            class="button tertiary"
            href="/"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            class="button tertiary"
            href="/"
          >
            <test-file-stub />
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
